dbms/src/Storages/DeltaMerge/ReadThread/ColumnSharingCache.cpp

```diff
@@ -61,6 +61,14 @@ void DMFileReaderPool::set(DMFileReader & from_reader, int64_t col_id, size_t st
     }
 }
 
+// Check is there any concurrent DMFileReader with `from_reader`.
+bool DMFileReaderPool::hasConcurrentReader(DMFileReader & from_reader)
+{
+    std::lock_guard lock(mtx);
+    auto itr = readers.find(from_reader.path());
+    return itr != readers.end() && itr->second.size() >= 2;
```

Comment:
So actually, there is a chance that the reader becomes a `ConcurrentReader` after `hasConcurrentReader` is called?